cpp/src/gandiva/filter.cc

```diff
@@ -18,13 +18,16 @@
 #include "gandiva/filter.h"
 
 #include <memory>
+#include <thread>
```

Comment:
These includes are also not necessary. Can you review all the includes in the PR.

Reply:
Done!